command/agent/acl_endpoint.go

```diff
@@ -138,8 +138,18 @@ func (s *HTTPServer) ACLTokenBootstrap(resp http.ResponseWriter, req *http.Reque
 		return nil, CodedError(405, ErrInvalidMethod)
 	}
 
-	// Format the request
-	args := structs.ACLTokenBootstrapRequest{}
+	var args structs.ACLTokenBootstrapRequest
+
+	if req.ContentLength == 0 {
+		args = structs.ACLTokenBootstrapRequest{
+			BootstrapSecret: "",
+		}
+	} else {
```

Comment:
You can rely on Go initializing structs to their zero/empty values and simplify this code a bit. No big deal.

```suggestion
	args := structs.ACLTokenBootstrapRequest{}

	if req.ContentLength != 0 {
```

Reply:
Simplify Code